mysql-test/suite/sys_vars/t/innodb_buffer_pool_dump_abort_loads.test

```diff
@@ -0,0 +1,130 @@
+#
+# MDEV-11455 - add status variable innodb_buffer_pool_load_abort
+#
+--source include/have_innodb.inc
+--source include/have_debug_sync.inc
+
+--echo
+--echo # innodb_buffer_pool_load_incomplete defaults 0
+SELECT variable_name, variable_value
+    FROM information_schema.global_status
+    WHERE LOWER(variable_name) = 'innodb_buffer_pool_load_incomplete';
+
+--echo
+--echo # populate with data
+
+CREATE TABLE t1 (
+c01 blob, c02 blob, c03 blob, c04 blob, c05 blob,
+c06 blob, c07 blob, c08 blob, c09 blob, c10 blob,
+c11 blob, c12 blob, c13 blob, c14 blob, c15 blob,
+c16 blob, c17 blob, c18 blob, c19 blob, c20 blob,
+c21 blob, c22 blob, c23 blob, c24 blob, c25 blob,
+c26 blob, c27 blob, c28 blob, c29 blob, c30 blob,
+c31 blob, c32 blob, c33 blob, c34 blob, c35 blob,
+c36 blob, c37 blob, c38 blob, c39 blob, c40 blob,
+c41 blob, c42 blob, c43 blob, c44 blob, c45 blob,
+c46 blob, c47 blob, c48 blob, c49 blob, c50 blob,
+c51 blob, c52 blob, c53 blob, c54 blob, c55 blob,
+c56 blob, c57 blob, c58 blob, c59 blob, c60 blob,
+c61 blob, c62 blob, c63 blob, c64 blob
+) ROW_FORMAT=dynamic;
+
+SET @a = repeat('a', 16 * 1024);
+INSERT INTO t1 VALUES (@a,@a,@a,@a,@a,
+@a,@a,@a,@a,@a,
+@a,@a,@a,@a,@a,
+@a,@a,@a,@a,@a,
+@a,@a,@a,@a,@a,
+@a,@a,@a,@a,@a,
+@a,@a,@a,@a,@a,
+@a,@a,@a,@a,@a,
+@a,@a,@a,@a,@a,
+@a,@a,@a,@a,@a,
+@a,@a,@a,@a,@a,
+@a,@a,@a,@a,@a,
+@a,@a,@a,@a
+);
+
+SET GLOBAL innodb_buffer_pool_dump_at_shutdown=1, GLOBAL innodb_fast_shutdown=0;
+# Dump on shutdown has been set - shutdown now
+
+--echo
+--echo # Restart server
+--source include/restart_mysqld.inc
+
+SET debug_sync="buf_load WAIT_FOR shutdown",
+    GLOBAL innodb_buffer_pool_load_now=1,
+    GLOBAL innodb_buffer_pool_dump_at_shutdown=1;
```

Comment:
I believe that the `SET DEBUG_SYNC` here is not having any effect on the `DEBUG_SYNC_C("buf_load")` in `buf_load()` that is executed by `buf_dump_thread`, which is not the same thread that executed the update callback `buffer_pool_load_now()`.
It would be better design to have the `buffer_pool_load_now()` perform the operation and not return to the client before the loading has completed.

If the DEBUG_SYNC had an effect, I believe that the server restart would hang in shutdown, because there is no `SET DEBUG_SYNC='… SIGNAL shutdown'` before the server restart. (I am assuming that the signal name `shutdown` has no magic associated with it.)

I do not have good ideas how to fix this. Maybe add a debug-only variable that would cause `buf_load()` to pause? Or a debug-only variable that would set `innodb_buffer_pool_load_abort=1` after loading one page?